bin/alluxio-start.sh

```diff
@@ -515,11 +515,26 @@ main() {
       start_proxies
       ;;
     local)
-      ALLUXIO_MASTER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
-      ALLUXIO_WORKER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
-      ALLUXIO_PROXY_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
-      ALLUXIO_JOB_MASTER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
-      ALLUXIO_JOB_WORKER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
+      local master_hostname=$(${BIN}/alluxio getConf ${ALLUXIO_MASTER_JAVA_OPTS}\
+                              alluxio.master.hostname)
+      if [[ -n ${master_hostname} ]]; then
+        local local_addresses=( "localhost" "127.0.0.1" $(hostname -s) $(hostname -f) )
+        if [[ $(uname -a) != Darwin* ]]; then
+          # Assuming Linux
+          local_addresses+=( $(hostname --ip-address) )
+        fi
+        local is_master_local=false
+        for local_address in ${local_addresses[*]}
+        do
+           if [[ ${local_address} == ${master_hostname} ]]; then
+             is_master_local=true
+             break
+           fi
+        done
+        if [[ ${is_master_local} == false ]]; then
+          echo "alluxio.master.hostname=localhost" >> "${ALLUXIO_CONF_DIR}/alluxio-site.properties"
+        fi
+      fi
```

Comment:
Shouldn't we append if the master hostname is unset?

Reply:
good catch, updated